dash/include/dash/allocator/CollectiveAllocator.h

```diff
@@ -123,12 +126,15 @@ class CollectiveAllocator
   /**
    * Move-assignment operator.
    */
-  self_t & operator=(const self_t && other) noexcept
+  self_t & operator=(self_t && other) noexcept
   {
     // Take ownership of other instance's allocation vector:
     if (this != &other) {
-      std::swap(_allocated, other._allocated);
```

Comment:
The use of `std::swap` should be OK here because `other` is slated for deallocation after a move, thus deallocating the current `_allocated`. Also, the call to `other._allocated.clear()` should not be required afaics.